harness/tests/determined/common/experimental/test_experiment.py

```diff
@@ -209,3 +210,38 @@ def test_unarchive_doesnt_update_local_on_rest_failure(
         raise AssertionError("bindings API call should raise an exception")
     except bindings.APIHttpError:
         assert expref.archived is None
+
+
+@responses.activate
+@mock.patch("determined.common.api.bindings.put_PutExperimentLabel")
+def test_add_label_url_encodes_label(
+    mock_bindings: mock.MagicMock,
+    make_expref: Callable[[int], experiment.Experiment],
+) -> None:
+    expref = make_expref(1)
+    label = "label with @#$%^&* symbols"
+    url_encoded_label = "label%20with%20%40%23%24%25%5E%26%2A%20symbols"
+    assert parse.unquote(url_encoded_label) == label
+
+    expref.add_label(label=label)
+
+    assert mock_bindings.call_count == 1
```

Comment:
I dunno... I get that this is how the code should work, but the assertion just doesn't seem relevant to the test. My own opinion here is that if you think "one call to bindings" is worth testing, then I think it should be its own test.

Reply:
i think you're right. i added them more out of anxiety ("but what if?") than practicality. 😄  removed